ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -210,6 +218,8 @@ StreamInfo get(Key key) {
 
   private final ExecutorService executorService;
 
+  private final List<ExecutorService> streamExecutors = new ArrayList<>();
+
```

Comment:
After RATIS-1126, we may simply use. NettyServerStreamRpc.executorService.  Or, we may add another 
```
private final ExecutorService streamExecutor;
```
for the stream write() and close().

Reply:
@szetszwo It will cause error.  Because one stream must use the same thread to write, if we use NettyServerStreamRpc.executorService, it will happen multi-thread call the write belongs to the same stream, error will happen. Then we must add synchronized to write method, which is not appropriate, because it will cause thread wait for lock .

Reply:
> Because one stream must use the same thread to write, ...

No.  We should not require using the same thread.  This is not the async model.

Think about this: for any moment of time, our program must guarantee only one thread calling channel.write(buffer).  Therefore, the program must be synchronise somewhere.  So, we may safely use a thread-pool (with multiple threads).

Reply:
>  for any moment of time, our program must guarantee only one thread calling channel.write(buffer)

@szetszwo Current implementation can not guarantee only one thread calling channel.write(buffer) at the same time.

Reply:
@szetszwo Could you help review this comment ?